### PR TITLE
Api documentation generation

### PR DESCRIPTION
**Pull request by gsouf**
This PR includes the implementation for the `zephir api` command.
# API generation

The core of the generation is ready, though there is still a few work to do, we can already use it.

The api generation is not hard coded. Instead there is core generator, in whitch we can plug a theme (a bit like sphinx documentation works). Right now the theme path is relative to zephir path, it's already planned to make this path also relative to the project root.
# Default Theme for api

The PR also includes a default theme for the api generation. The theme is not 100% ready, there is still a couple of htlm + css to do to finish it.
# Docblock parser

The PR also includes a docblock parser, that was needed to get the annotations.   The docblock parsed is not 100% ready, but it returns a valid Docblock object. The api should not change, we can already safely use it and rewrite the internal work of the parser latter.
# Tests

Right now there is no test written, it's already planed to write them

### CODE REVIEW

File: nbproject/project.properties [hunk 1]
```diff
@@ -0,0 +1,7 @@
+include.path=${php.global.include.path}
```
Comment (ovr) [inline nbproject/project.properties]: I think nb project is not needed

**Comment by ovr** [comment]
Merge is not a good way
Read about `git rebase`
And rebase it

File: nbproject/project.properties (hunk 1 above)
Comment (SneakyBobito) [inline nbproject/project.properties]: !!!!yes indeed, going to remove it

**Comment by SneakyBobito** [comment]
Sounds better like that !

**Comment by ovr** [comment]
cc @andresgutierrez @nkt @andresgutierrez 
Opinion?

**Comment by phalcon** [comment]
Thanks!

**Comment by SneakyBobito** [comment]
Nice to see it merged ! I still have to provide docblocks parsing before going further

**Comment by mruz** [comment]
@gsouf can I access to `version` from config.json in the main layout?

**Comment by gsouf** [comment]
Hi @mruz 

This is not the best option (and i highly deprecat it), but it's the only one you have right now : 

``<?php echo $this->themeOption('version'); ?>``

See the example here : https://github.com/phalcon/zephir/blob/master/templates/Api/themes/zephir/layout/layout.phtml#L40

An update is on the way. I will improve the access to variables like the version

**Comment by mruz** [comment]
@gsouf OK thanks, I'm waiting for the update.
I have one more question, how to display docblock for the classes? I did try:
```php
<?= $classDefinition->getParsedDocBlock()->getDescription() ?>
```

**Comment by gsouf** [comment]
@mruz When I writted the generator Zephir was missing the feature of the class docblock. Unfortunately i didnt have the time to learn how the parser works. 

@phalcon what's about the class docblock parsing ?

**Comment by gsouf** [comment]
@mruz In the master branch, method ``projectConfig`` is available now in templates, see the example : 

https://github.com/phalcon/zephir/blob/master/templates/Api/themes/zephir/layout/layout.phtml#L43

**Comment by mruz** [comment]
@gsouf thanks!

**Comment by mruz** [comment]
@gsouf stubs already parsing docblock https://github.com/phalcon/zephir/issues/713

**Comment by gsouf** [comment]
@mruz it doesnt fit the needings

**Comment by gsouf** [comment]
Now class docblock are available : 

``compilerFile->getClassDefinitiion()->getDocblock()`` or ``compilerFile->getClassDefinitiion()->getParsedDocblock()``

**Comment by mruz** [comment]
In the `themes/ice/partials/class/content.phtml` I'm using:
```php
<?php $docblock = $classDefinition->getParsedDocBlock(); ?>
```
and
```php
<?php if ($docblock) { ?>
    <div class="text-muted">
    <?= $docblock->getSummary() ?>
    </div>
    <div>
    <?= var_dump($docblock) ?>
    <?= $docblock->getDescription() ?>
    </div>
<?php } ?>
```

And now it works ;) but there is no spaces in the docs:

> $validation=newIce\Validation();$validation->rules(['fullName'=>'required','email'=>'required|email','repeatEmail'=>'same:email','about'=>'required|length:10,5000',]);$valid=$validation->validate($_POST);if(!$valid){$messages=$validation->getMessages();}

@gsouf @ovr any idea?

**Comment by gsouf** [comment]
You mean no new line ?  In your css use ´´white-space : pre´´

**Comment by mruz** [comment]
Yes, no new lines and no spaces. In the source code:

```html
<pre><code>$validation=newIce\Validation();$validation->rules(['fullName'=>'required','email'=>'required|email','repeatEmail'=>'same:email','about'=>'required|length:10,5000',]);$valid=$validation->validate($_POST);if(!$valid){$messages=$validation->getMessages();}</code></pre>
```

**Comment by mruz** [comment]
I found it occurs if some tags exists, eg. `@package`:
```
 * @package     Ice/Validation
 *
 * <pre><code>
 *  $validation = new Ice\Validation();
 *  
 *  $validation->rules([
 *      'fullName' => 'required',
 *      'email' => 'required|email',
 *      'repeatEmail' => 'same:email',
 *      'about' => 'required|length:10,5000',
 *  ]);
 *  
 *  $valid = $validation->validate($_POST);
 *  
 *  if (!$valid) {
 *      $messages = $validation->getMessages();
 *  }
 * </code></pre>
 */
```

**Comment by gsouf** [comment]
I'll take care of this, just give me some time, i'll be very busy this week.

**Comment by mruz** [comment]
@gsouf you can see it at my theme. I added [docblock](https://github.com/ice/doc/blob/master/themes/ice/partials/class/content.phtml#L36), but it looks like in http://doc.iceframework.org/latest/class/Ice/Validation.html. If I remove tags `@package`, `category`, etc. from the source http://doc.iceframework.org/latest/source/Ice/Validation.html it generates properly with spaces.

**Pull request merged**